gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java

```diff
@@ -0,0 +1,249 @@
+   *
+   * @return {@code true} if this disk exists, {@code false} otherwise
+   * @throws ComputeException upon failure
+   */
+  public boolean exists() {
+    return reload(DiskOption.fields()) != null;
+  }
+
+  /**
+   * Fetches current disk's latest information. Returns {@code null} if the disk does not exist.
+   *
+   * @param options disk options
+   * @return a {@code Disk} object with latest information or {@code null} if not found
+   * @throws ComputeException upon failure
+   */
+  public Disk reload(DiskOption... options) {
+    return compute.get(diskId(), options);
+  }
+
+  /**
+   * Deletes this disk.
+   *
+   * @return a zone operation if the delete request was successfully sent, {@code null} if the disk
+   *     was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation delete(OperationOption... options) {
+    return compute.delete(diskId(), options);
+  }
+
+  /**
+   * Creates a snapshot for this disk given the snapshot's name.
+   *
+   * @return a zone operation if the snapshot creation was successfully requested. {@code null} if
+   *     the disk was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation createSnapshot(String snapshot, OperationOption... options) {
+    return compute.create(SnapshotInfo.of(SnapshotId.of(snapshot), diskId()), options);
+  }
+
+  /**
+   * Creates a snapshot for this disk given the snapshot's name and description.
+   *
+   * @return a zone operation if the snapshot creation was successfully requested. {@code null} if
+   *     this disk was not found
+   * @throws ComputeException upon failure
+   */
+  public Operation createSnapshot(String snapshot, String description, OperationOption... options) {
+    SnapshotInfo snapshotInfo = SnapshotInfo.builder(SnapshotId.of(snapshot), diskId())
+        .description(description)
+        .build();
+    return compute.create(snapshotInfo, options);
+  }
+
+  /**
+   * Creates an image for this disk given the image's name.
+   *
+   * @return a global operation if the image creation was successfully requested
+   * @throws ComputeException upon failure
+   */
+  public Operation createImage(String image, OperationOption... options) {
+    ImageInfo imageInfo = ImageInfo.of(ImageId.of(image), DiskImageConfiguration.of(diskId()));
+    return compute.create(imageInfo, options);
+  }
+
+  /**
+   * Creates an image for this disk given the image's name and description.
+   *
+   * @return a global operation if the image creation was successfully requested
+   * @throws ComputeException upon failure
+   */
+  public Operation createImage(String image, String description, OperationOption... options) {
+    ImageInfo imageInfo = ImageInfo.builder(ImageId.of(image), DiskImageConfiguration.of(diskId()))
+        .description(description)
+        .build();
+    return compute.create(imageInfo, options);
+  }
+
+  /**
+   * Returns the disk's {@code Compute} object used to issue requests.
```

Comment:
I didn't use `@link` because Compute should be already linked in the method's javadoc as the return type.